### PR TITLE
Migrate to poetry and package for nix

### PR DESCRIPTION
**Pull request by FugiMuffi**
This PR implements the following changes:
- Migrate the project to poetry
- Remove Docker config (it would basically have to be recreated from scratch, if we need it)
- Merge email configuration into settings.py
- Package the project using nix and provide a NixOS module

[The groundwork](https://github.com/sbruder/course-management/tree/nix) was laid by @sbruder and major parts of it were carried over and expanded by me.

Thorough testing has yet to be conducted and any help or feedback is much appreciated.

### CODE REVIEW

File: module.nix [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+  };
+in
+{
+  options.services.course-management = with lib; {
+    enable = mkEnableOption "iFSR course management";
+    package = mkOption {
+      type = types.package;
+      default = pkgs.course-management;
+      description = "The package to use.";
+    };
```
Comment (therealr5) [inline module.nix:119]: Maybe it's worthwile introducing a `user` option in here that we can use for sops.

File: module.nix [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+          SystemCallArchitectures = "native";
+          SystemCallFilter = "@system-service";
+        };
+      };
+
+      services.nginx = lib.mkIf (cfg.hostName != null) {
+        enable = true;
+        recommendedProxySettings = lib.mkDefault true;
+
+        virtualHosts.${cfg.hostName} = {
```
Comment (therealr5) [inline module.nix:342]:
```suggestion
        virtualHosts.${cfg.hostName} = {
          enableACME = true;
          forceSSL = true;
```
Could also be put behind another option. Not entirely sure about that.

Comment (FugiMuffi) [reply module.nix:360]: I'd rather configure TLS downstream, this way it's simpler to use different configurations, e.g. a wildcard cert.

**Review by therealr5 — APPROVED** [review]

**Comment by therealr5** [comment]
We sadly have some merge conflicts due to da9f7bd :(

**Comment by FugiMuffi** [comment]
The conflicts are because I moved everything into a subdir, it should be no a problem to resolve them

**Comment by FugiMuffi** [comment]
Github: "These conflicts are too complex to resolve in the web editor :("
git merge: ¯\\\_(ツ)\_/¯

**Pull request merged**